src/node.cc

```diff
@@ -2192,7 +2192,7 @@ static void WaitForInspectorDisconnect(Environment* env) {
   if (env->inspector_agent()->IsConnected()) {
     // Restore signal dispositions, the app is done and is no longer
     // capable of handling signals.
-#ifdef __POSIX__
+#if defined(__POSIX__) && !defined(NODE_SHARED_MODE)
```

Comment:
I didn't think the inspector applied any additional signal handlers therefore it seems wrong to squash them all on the exit of the node runtime when loaded as a shared library (potentially someone could shutdown the node runtime within their own application and I wouldn't want this to squash any signal handlers in the caller).

I'm happy to be corrected though if there was a separate reason for this loop to be added in here. @eugeneo added this sequence in https://github.com/nodejs/node/commit/6626919205fa71628e362646a9d94aade2fa8332 so may have more info and be able to comment.

To be fair, as far as I can see this is only invoked when someone calls process.exit() from within javascript, in which case it'll currently be taking the wrapping process down anyway. on that basis I suspect it won't make a lot of practical difference either way, although I'd prefer to be sure that we really didn't want to squash the handlers in this "embedded" case before removing it.